### ISSUE TITLE
Hydrogen in Song mode and as Jack slave skipping first beat (and possibly others)

### ISSUE THREAD

**Issue by blablack** [hydrogen-music/hydrogen#379]
Hello all,

Has anybody else encountered in the latest Hydrogen git (29/05/2016) is skipping beats when being a jack slave (master being Bitwig in my case) and in song mode?

I have a 4 patterns song that I'm looping and Hydrogen would skip sometimes the first beats of the 4 patterns loop (let say if looping it 4 times, it would skip it 3 times) and it would skip random beats in the loop. It started to happen very recently (wasn't happening in April, is now happening in May).

As Jack master or in pattern mode, everything runs smoothly.

Anybody else?

**Comment by mauser** [comment]
Hi! 

Don't know much about our jack code, but there was an bugfix recently which fixed the clock when working in song mode with jack enabled: https://github.com/hydrogen-music/hydrogen/issues/369 . Could you try if this fix broke the transport for you?

**Comment by blablack** [comment]
Hi,

Just tested it but the issue persist - Hydrogen is skipping the first beat
when it is not master.
The issue seem to be with Bitwig though as using just QjactCTL to
start/stop Hydrogen I cannot reproduce the issue.

I need to test with another master to see if it happens again.

**Comment by mauser** [comment]
Hi! 

If it hasn't happened in April, then it is likely that the fix for #369 introduced this problem, since there are no other audio engine related commits in this range.. 

I will see if i can reproduce the problem here with a demo version of bitwig.

**Comment by mauser** [comment]
I can sometimes reproduce this problem. I couldn't reproduce it everytime with bitwig (1.3.11), but with [klick](http://das.nasophon.de/klick/).  The following observations were made with `klick -T 4/4 120`, so running it as the jack time master with a 4/4 bar and 120bpm. 

Using pattern mode , hydrogen skipped always the first beat. Using song mode, hydrogen skipped the first beat if the song has reached its end and hydrogen started again at position 0. This seems to be similiar to #202. In contrast to the observation of Aurelien, i can't observe any random beat skips. It's always the first beat in the pattern.. Hydrogen outputs this:

`(E) Sampler::__render_note Note pos in the future?? Current frames: 229, note frame pos: 88200`

But if i go back to a revision where #369 wasn't fixed yet, everything works on my side. No beats are skipped.  @xjjx , Pawel, could you have a look here?

**Comment by mauser** [comment]
@blablack : Btw, on my system, bitwig is not able to set the tempo of the jack transport if it is used as time master. Does this work for you?

**Comment by blablack** [comment]
@mauser Yes it does work for me - Bitwig would send the transport tempo.
Regarding Hydrogen, it actually works if Bitwig is NOT set as transport
master.

To me after the testing I have done it looks like the issue is with Bitwig
itself - I don't user any other software synced with bitwig but Hydrogen to
judge if the behaviour affects other softwar.

In my opinion, this issue can be close.

**Comment by xjjx** [comment]
I'll check this asap.

**Comment by thijz** [comment]
in the past i have run into various issues with sync and/or jack transport master/slave related things
there are a number of tickets about this (will see if i can find them and add them here for reference)

**Comment by blablack** [comment]
Hi all,

I noticed Bitwig behaving the same with other software when set as Master.
I think this issue can be closed, it is not related to Hydrogen but to Bitwig.

**Comment by IvanDSM** [comment]
Don't know if it's the same thing, but when I have Ardour as a Time Master and I start playing a bar, it doesn't play the first beat, only what's after that.

**Comment by layk** [comment]
This is now happening to me with seq24 or sequencer64 (https://github.com/ahlstromcj/sequencer64) as master with 0.7.9. The first beat does not play correctly.
I think my previous fedora installation had 0.9.6 installed and everything was fine. I tried to compile my self 0.9.7 from git but to no avail.
Compiling 0.9.6.1 works correctly.

**Comment by xjjx** [comment]
The last time I tried to solve the problem I can't reproduce it, but maybe this time will be better ;-)

Could you compile this tool:
https://sourceforge.net/p/nxjsm/code/HEAD/tree/trunk/
make nxjsm_jt_test

... and make simple test:
Start it from console before you run hydrogen and seq24 like this:
nxjsm_jt_test | tee jt_test.log

And while it's running try to reproduce the problem.
Then attach jt_test.log or you can send me to xj@wp.pl

**Comment by layk** [comment]
@xjjx Log file attached.
Starting hydrogen from console gives me following :
(E) Effects::getPluginList Error loading the library. (/usr/lib/ladspa/ladspa-rubberband.so)
(E) Effects::getPluginList Error loading the library. (/usr/lib64/ladspa/calf.so)
And every time the first beat errors it gives me:
(E) Sampler::__render_note Note pos in the future?? Current frames: 250, note frame pos: 96000
(E) Sampler::__render_note Note pos in the future?? Current frames: 250, note frame pos: 96000
I tested the version that's in Fedora 25 repositories 0.9.7

[jt_test.log.zip](https://github.com/hydrogen-music/hydrogen/files/888759/jt_test.log.zip)

**Comment by nextl00p** [comment]
Also have the issue. The first beet seems to get skipped. I use ardour as jack time master.

```
(E) Sampler::__render_note Note pos in the future?? Current frames: 765, note frame pos: 58800
(E) Sampler::__render_note Note pos in the future?? Current frames: 336, note frame pos: 58800
(E) Sampler::__render_note Note pos in the future?? Current frames: 926, note frame pos: 58800
(E) Sampler::__render_note Note pos in the future?? Current frames: 497, note frame pos: 58800
 (E) Sampler::__render_note Note pos in the future?? Current frames: 61, note frame pos: 58800
```

**Comment by layk** [comment]
Just noticed today that running 0.9.7 from my fedora installation the problem disappears after following steps.
1. Initially open hydrogen and make it listen to transport with J.TRANS button. Starting transport from another app the first beat is skipped.
2. In hydrogen make hydrogen as master with master button and disable it.
3. I use seq64 as transport master so i go to Jack Sync options and do Jack Transport Disconnect/connect with transport master option enabled and hydrogen does not skip the first beat.

So maybe this works for others? Start hydrogen and enable master/disable master and then launch the real transport master program.

This leads to thinking that maybe pushing J.MASTER button initializes something that is not correct after launching the program.

**Comment by trebmuh** [comment]
`(I've changed the title to ease when searching through the issues. I hope you don't mind @blablack)`

**Comment by theGreatWhiteShark** [comment]
This issue should be fixed in #678

**Comment by mauser** [comment]
Closed: as @theGreatWhiteShark mentioned, the issue seems to be fixed with Hydrogen 1.0.
